According to http://doc.qt.io/qt-5/qkeysequence.html:
StandardKey.Cut under KDE includes "Ctrl+X, F20, Shift+Del", Obviously it's not match shortcut scheme of KDE.
Here we use: Ctrl+X for cut, Shift+Del for Delete directly.
diff -Nur plasma-desktop-5.4.3/containments/desktop/package/contents/ui/FolderView.qml plasma-desktop-5.4.3key/containments/desktop/package/contents/ui/FolderView.qml
--- plasma-desktop-5.4.3/containments/desktop/package/contents/ui/FolderView.qml	2015-11-05 20:44:49.000000000 +0800
+++ plasma-desktop-5.4.3key/containments/desktop/package/contents/ui/FolderView.qml	2015-12-23 23:06:12.535035377 +0800
@@ -670,6 +670,16 @@
                     } else if (event.key == Qt.Key_End) {
                         currentIndex = count - 1;
                         updateSelection(event.modifiers);
+                    } else if (event.matches(StandardKey.Copy)) {
+                        dir.copy();
+                    } else if (event.matches(StandardKey.Paste)) {
+                        dir.paste();
+                    } else if (event.modifiers & Qt.ControlModifier) {
+                        if (event.key == Qt.Key_X)
+                            dir.cut();
+                    } else if (event.modifiers & Qt.ShiftModifier) {
+                        if (event.key == Qt.Key_Delete)
+                            dir.deleteSelected();
                     }
                 }
 
diff -Nur plasma-desktop-5.4.3/containments/desktop/plugins/folder/foldermodel.h plasma-desktop-5.4.3key/containments/desktop/plugins/folder/foldermodel.h
--- plasma-desktop-5.4.3/containments/desktop/plugins/folder/foldermodel.h	2015-11-05 20:44:49.000000000 +0800
+++ plasma-desktop-5.4.3key/containments/desktop/plugins/folder/foldermodel.h	2015-12-23 23:05:56.792034665 +0800
@@ -191,6 +191,11 @@
         bool isDir(const QModelIndex &index, const KDirModel *dirModel) const;
         bool lessThan(const QModelIndex &left, const QModelIndex &right) const;
 
+        Q_INVOKABLE void paste();
+        Q_INVOKABLE void copy();
+        Q_INVOKABLE void cut();
+        Q_INVOKABLE void deleteSelected();
+
     Q_SIGNALS:
         void urlChanged() const;
         void resolvedUrlChanged() const;
@@ -220,13 +225,9 @@
         void statResult(KJob *job);
         void evictFromIsDirCache(const KFileItemList &items);
         void selectionChanged(QItemSelection selected, QItemSelection deselected);
-        void copy();
-        void cut();
-        void paste();
         void pasteTo();
         void refresh();
         void moveSelectedToTrash();
-        void deleteSelected();
         void emptyTrashBin();
         void undoTextChanged(const QString &text);
 
